integration-testing/build.gradle.kts

```diff
@@ -0,0 +1,87 @@
+import org.jetbrains.kotlin.gradle.utils.NativeCompilerDownloader
+
+plugins {
+    kotlin("jvm")
+}
+
+repositories {
+    mavenLocal()
+    maven("https://maven.pkg.jetbrains.space/kotlin/p/kotlin/dev")
+    mavenCentral()
+}
+
+java {
+    toolchain {
+        languageVersion.set(JavaLanguageVersion.of(11))
+    }
+}
+
+kotlin {
+    jvmToolchain(11)
+}
+
+val kotlin_version = providers.gradleProperty("kotlin_version").orNull
+val atomicfu_snapshot_version = providers.gradleProperty("version").orNull
+
+sourceSets {
+    create("mavenTest") {
+        compileClasspath += files(sourceSets.main.get().output, configurations.testRuntimeClasspath)
+        runtimeClasspath += output + compileClasspath
+    }
+
+    create("functionalTest") {
+        compileClasspath += files(sourceSets.main.get().output, configurations.testRuntimeClasspath)
+        runtimeClasspath += output + compileClasspath
+    }
+}
+
+dependencies {
+    // common dependencies
+    implementation("org.jetbrains.kotlin:kotlin-stdlib-jdk8")
+    testImplementation("org.jetbrains.kotlin:kotlin-test")
+    implementation("org.jetbrains.kotlin:kotlin-script-runtime")
+    implementation(kotlin("script-runtime"))
+
+    // mavenTest dependencies
+    "mavenTestImplementation"("org.jetbrains.kotlinx:atomicfu-jvm:$atomicfu_snapshot_version")
+
+    // functionalTest dependencies
+    "functionalTestImplementation"(gradleTestKit())
+    "functionalTestApi"("org.ow2.asm:asm:9.3")
+    "functionalTestApi"("org.ow2.asm:asm-commons:9.3")
+}
+
+val mavenTest by tasks.registering(Test::class) {
+    testClassesDirs = sourceSets["mavenTest"].output.classesDirs
+    classpath = sourceSets["mavenTest"].runtimeClasspath
+    
+    dependsOn(":atomicfu:publishToMavenLocal")
+}
+
+val functionalTest by tasks.registering(Test::class) {
```

Comment:
It seems like there's a problem with the task result's caching.

What I did locally:
- checked out this PR;
- executed `:integration-testing:check`, all tests passed
- broke `AtomicFUGradlePlugin` by commenting out dependencies inclusion;
- executed `:integration-testing:check` once again and this time, there were no failures too.
Only by performing `clean` and then running the `check` again I was able to get test failures.

I think we can plug `outputs.upToDateWhen { false }` for now and address the issue later in https://github.com/Kotlin/kotlinx-atomicfu/issues/362

Reply:
Thank you, I didn't think of that
Fixed for both mavenTest and functionalTest, and mentioned the problem in #362